builtin/providers/aws/resource_aws_kinesis_analytics_test.go

```diff
@@ -0,0 +1,287 @@
+package aws
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/kinesisanalytics"
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+func TestAccAWSKinesisAnalytics_basic(t *testing.T) {
+	var desc kinesisanalytics.ApplicationDetail
+
+	rInt := acctest.RandInt()
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: destroyKinesisAnalytics,
+		Steps: []resource.TestStep{
+			{
+				Config: kinesisAnalyticsBasicCreateConfig(rInt),
+				Check: resource.ComposeTestCheckFunc(
+					doesKinesisAnalyticsInstanceExist("aws_kinesis_analytics.test_application", &desc),
+					areRootAttributesCorrect(&desc),
+				),
+			},
+		},
+	})
+}
+
+func TestAccAWSKinesisAnalytics_stream_connections(t *testing.T) {
+	var desc kinesisanalytics.ApplicationDetail
+
+	rInt := acctest.RandInt()
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: destroyKinesisAnalytics,
+		Steps: []resource.TestStep{
+			{
+				Config: kinesisAnalyticsCreateWithStreamsConfig(rInt),
+				Check: resource.ComposeTestCheckFunc(
+					doesKinesisAnalyticsInstanceExist("aws_kinesis_analytics.test_application", &desc),
+					areInputStreamAttributesCorrect(&desc),
+					areOutputStreamAttributesCorrect(&desc),
+				),
+			},
+		},
+	})
+}
+
+func doesKinesisAnalyticsInstanceExist(n string, desc *kinesisanalytics.ApplicationDetail) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("Not found: %s", n)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("No Kinesis Application ID is set")
+		}
+
+		conn := testAccProvider.Meta().(*AWSClient).kinesisanalyticsconn
+		describeOpts := &kinesisanalytics.DescribeApplicationInput{
+			ApplicationName: aws.String(rs.Primary.Attributes["name"]),
+		}
+
+		resp, err := conn.DescribeApplication(describeOpts)
+		if err != nil {
+			return err
+		}
+
+		*desc = *resp.ApplicationDetail
+
+		return nil
+	}
+}
+
+func areRootAttributesCorrect(desc *kinesisanalytics.ApplicationDetail) resource.TestCheckFunc {
```

Comment:
This will need renamed to be much more local scope and include AWSKinesisAnalytics